web/modules/custom/lit_cover_service/src/Service/CoverService.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+
+namespace Drupal\lit_cover_service\Service;
+
+use Drupal\Core\File\FileSystemInterface;
+use Drupal\file\Entity\File;
+use Drupal\file\FileInterface;
+use Drupal\lit_cover_service\OpenPlatform\TokenClient;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+
+/**
+ * Class CoverService
+ */
+class CoverService {
+
+  private const COVER_SERVICE_URL = 'https://cover.dandigbib.org/api/v2/covers';
+  private const DRUPAL_FILE_PATH = 'public://cover_dandigbib_org';
```

Comment:
Why the home made client? Why is https://github.com/danskernesdigitalebibliotek/ddb-cover-service-php-client not used?

Reply:
Because the API is dead simple. Only the `getCoverUrlForIsbn()` methods is covered by the DDB Client. The rest, checking if we have the cover loaclly, downloading the file, get credentials from the existing open platform module would still need to be handled by this module.

Reply:
Sorry.... but I will block this PR until the client library is used. It's maybe simple but the client contains the API version etc. and is dead simple to use..... with your argument... we should never have made a client library... which is just wrong.

```
try {
    $config = _ting_cover_ddb_get_client_config();
    $apiInstance = new CoverApi(
      new Client(),
      $config
    );
    _ting_covers_ddb_debug_log('Requested covers for pids: ' . implode(', ', array_keys($entities)));
    $retrieved = $apiInstance->getCoverCollection('pid', array_keys($entities), array('original', 'large'));
  }
  catch (Exception $e) {
    watchdog('ting_covers_ddb', 'Unable to retrieve covers: %message', array('%message' => $e->getMessage()), WATCHDOG_ERROR);
    return $covers;
  }
```

https://github.com/ding2/ding2/blob/master/modules/ting_covers_ddb/ting_covers_ddb.module#L105

It's way simpler to use the client than have it manual coded here...